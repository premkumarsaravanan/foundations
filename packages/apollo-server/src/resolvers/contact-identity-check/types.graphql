
# Types
type ContactIdentityDocument {
  typeId: String
  expiry: String
  details: String
  fileUrl: String
  _links: JSON
  _embedded: JSON
}

type ContactIdentityCheck {
  id: String
  contactId: String
  created: String
  modified: String
  checkDate: String
  status: String
  negotiatorId: String
  documents: [ContactIdentityDocument!]
  metadata: JSON
  _links: JSON
  _embedded: JSON
}

type ContactIdentityChecks {
  contactIdentityChecks: [ContactIdentityCheck!]
  pageNumber: Int
  pageSize: Int
  pageCount: Int
  totalCount: Int
  _links: JSON
}

# Input Types
input ContactIdentityDocumentInput {
  typeId: String
  expiry: String
  details: String
  fileUrl: String
}

input ContactIdentityCheckInput {
  checkDate: String
  status: String
  negotiatorId: String
  documents: [ContactIdentityDocumentInput!]
  metadata: JSON
}
