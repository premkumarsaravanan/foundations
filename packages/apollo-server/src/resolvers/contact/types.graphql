# Types
type ContactAddress {
  type: String
  buildingName: String
  buildingNumber: String
  line1: String
  line2: String
  line3: String
  line4: String
  postcode: String
  countryId: String
}

type ContactCommunication {
  label: String
  detail: String
}

type Contact {
  id: String
  created: String
  modified: String
  title: String
  forename: String
  surname: String
  dateOfBirth: String
  active: Boolean
  marketingConsent: String
  identityCheck: String
  communications: [ContactCommunication!]
  addresses: [ContactAddress!]
  officeIds: [String!]
  negotiatorIds: [String!]
  metadata: JSON
  _links: JSON
  _embedded: JSON
}

type Contacts {
  _embedded: [Contact!]
  pageNumber: Int
  pageSize: Int
  pageCount: Int
  totalCount: Int
  _links: JSON
}

# Input Types
input ContactCommunicationInput {
  label: String
  detail: String
}

input ContactAddressInput {
  type: String
  buildingName: String
  buildingNumber: String
  line1: String
  line2: String
  line3: String
  line4: String
  postcode: String
  countryId: String
}
