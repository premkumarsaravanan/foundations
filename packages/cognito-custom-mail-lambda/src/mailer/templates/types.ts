export interface ForgottenPasswordTemplateParams {
  verificationCode: string
  userName: string
  url: string
}

export interface ConfirmPasswordTemplateParams {
  userName: string
  url: string
}
