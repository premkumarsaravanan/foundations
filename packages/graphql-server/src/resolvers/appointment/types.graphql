# Types
type AppointmentContact {

  """
  The unique identifier of the contact
  """
  id: String

  """
  The name of the contact
  """
  name: String

  """
  The home phone number of the contact
  """
  homePhone: String

  """
  The work phone number of the contact
  """
  workPhone: String

  """
  The mobile phone number of the contact
  """
  mobilePhone: String

  """
  The email address of the contact
  """
  email: String
}

type AppointmentAttendee {
  """
  The unique identifier of the attendee
  """
  id: String

  """
  The type of attendee
  """
  type: String

  """
  A collection of contacts relating to the attendee
  """
  contacts: [AppointmentContact]
}

type AppointmentFollowUp {
  """
  The date when the appointment should be followed up
  """
  due: String

  """
  The unique identifier of a pre-defined follow up response type
  """
  responseId: String

  """
  Free text internal follow up notes to be stored against the appointment
  """
  notes: String
}

type Appointment {
  """
  The unique identifier of the appointment
  """
  id: String

  """
  The date and time when the appointment was created
  """
  created: String

  """
  The date and time when the appointment was last modified
  """
  modified: String

  """
  The date and time when the appointment will start
  """
  start: String

  """
  The date and time when the appointment will end
  """
  end: String

  """
  The unique identifier of the appointment type
  """
  typeId: String

  """
  A free text description about the appointment
  """
  description: String

  """
  A flag denoting whether or not the appointment recurs
  """
  recurring: Boolean

  """
  A flag denoting whether or not the appointment has been cancelled
  """
  cancelled: Boolean

  """
  Follow up information relating to the appointment
  """
  followUp: AppointmentFollowUp

  """
  The unique identifier of the property related to the appointment
  """
  propertyId: String

  """
  The unique identifier of the negotiator that organised the appointment
  """
  organiserId: String

  """
  A collection of unique identifiers of negotiators attached to the appointment
  """
  negotiatorIds: [String!]

  """
  A collection of unique identifiers of offices attached to the appointment
  """
  officeIds: [String!]

  """
  An appointment attendee
  """
  attendee: AppointmentAttendee

  """
  A flag denoting whether or not the appointment will be accompanied by one or more negotiators
  """
  accompanied: Boolean

  """
  A flag denoting whether or not the main negotiator has confirmed their attendance
  """
  negotiatorConfirmed: Boolean

  """
  A flag denoting whether or not the attendee has confirmed their attendance
  """
  attendeeConfirmed: Boolean

  """
  A flag denoting whether or not the property and/or property's vendor has confirmed their attendance
  """
  propertyConfirmed: Boolean

  """
  App specific metadata that has been set against the appointment
  """
  metadata: JSON

  """
  The ETag for the current version of the appointment. Used for managing update concurrency
  """
  _eTag: String
  _links: JSON
  _embedded: JSON
}

type GetAppointmentsResult {
  pageNumber: Int
  pageSize: Int
  pageCount: Int
  totalCount: Int
  _links: JSON,
  _embedded: [Appointment]
}


input UpdateAppointmentInputAttendee {
    """The unique identifier of the attendee"""
  id: String

  """The type of attendee (applicant/contact/landlord/tenant)"""
  type: String

  """
  A flag denoting whether or not the attendee has confirmed their attendance
  """
  confirmed: Boolean
}

# Input Types
input CreateAppointmentInputAttendee {
    """The unique identifier of the attendee"""
  id: String

  """The type of attendee (applicant/contact/landlord/tenant)"""
  type: String
}

input CreateAppointmentInputRecurrence {
    """The numeric value denoting how often the appointment will recur"""
  interval: Int

  """The type of unit that the `interval` applies to"""
  type: String

  """The date and time when the recurrence will stop"""
  until: String
}

input UpdateAppointmentInputRecurrence {
  """The type of unit that the `interval` applies to"""
  type: String

  """The numeric value denoting how often the appointment will recur"""
  interval: Int

  """The date and time when the recurrence will stop"""
  until: String
}

input UpdateAppointmentInputFollowUp {
  """The unique identifier of a pre-defined follow up response type"""
  responseId: String

  """The internal follow up notes to be stored against the appointment"""
  notes: String
}
