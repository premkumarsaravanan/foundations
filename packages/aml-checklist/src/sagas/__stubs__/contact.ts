export const contact = {
  id: 'MKC13000122',
  created: '2017-01-10T11:33:37.0000000Z',
  modified: '2020-01-21T14:04:50.0000000Z',
  title: 'Msss',
  forename: 'Holly',
  surname: 'Rees',
  dateOfBirth: '1986-06-30',
  active: true,
  marketingConsent: 'notAsked',
  source: {
    id: '',
    type: ''
  },
  homePhone: '01632 968608',
  workPhone: '020 7946 8608',
  mobilePhone: '07700 908608',
  email: 'hrees927@rpsfiction.net',
  primaryAddress: {
    type: 'home',
    buildingName: 'Larch Cottage',
    buildingNumber: '12',
    line1: 'High Street North',
    line2: 'Stewkley',
    line3: 'Leighton Buzzard',
    line4: 'Buckinghamshire',
    postcode: 'LU7 0EP',
    countryId: 'GB'
  },
  secondaryAddress: {
    type: 'previous',
    buildingName: 'Larch Cottage',
    buildingNumber: '7',
    line1: 'Ledburn',
    line2: 'Leighton Buzzard',
    line3: 'Buckinghamshire',
    line4: '',
    postcode: 'LU7 0PX',
    countryId: ''
  },
  officeIds: ['MKC'],
  negotiatorIds: ['JPB'],
  metadata: {
    addresses: [
      {
        documentImage:
          'https://reapit-dev-app-store-media.s3.eu-west-2.amazonaws.com/home-12-Larch Cottage-LU7 0EP.png',
        year: '2',
        month: '2',
        documentType: 'Bank / Building Society / Statement'
      },
      {
        documentImage:
          'https://reapit-dev-app-store-media.s3.eu-west-2.amazonaws.com/previous-7-Larch Cottage-LU7 0PX(1).jpg',
        year: '3',
        month: '1',
        documentType: 'Credit Card Statements from main provider'
      }
    ],
    declarationRisk: {
      reason: 'Test',
      type: 'Simplified',
      riskAssessmentForm:
        'https://reapit-dev-app-store-media.s3.eu-west-2.amazonaws.com/riskAssessment-Simplified-Test(1).png'
    }
  },
  _links: {
    self: {
      href: '/contacts/MKC13000122'
    },
    documents: {
      href: '/documents/?OwnerType=contact&OwnerId=MKC13000122'
    },
    identityChecks: {
      href: '/identityChecks/?ContactId=MKC13000122'
    }
  }
}

export const idCheck = {
  id: 'RPT19000083',
  contactId: 'MKC13000122',
  created: '0001-01-01T00:00:00',
  modified: '2020-01-21T03:39:53',
  checkDate: '0001-01-01T00:00:00',
  status: 'pass',
  negotiatorId: 'LJW',
  document1: {
    typeId: 'PP',
    expiry: '2020-07-03T23:00:00',
    details: '123123'
  },
  document2: {
    typeId: 'BB',
    expiry: '2020-01-12T00:00:00',
    details: 'ID Reference'
  },
  metadata: {
    primaryIdUrl: 'https://reapit-app-store-app-media.s3.eu-west-2.amazonaws.com/AYL19000001-testst.png',
    secondaryIdUrl: 'https://reapit-dev-app-store-media.s3.eu-west-2.amazonaws.com/MKC13000122-ID Reference.png'
  },
  _links: {
    self: {
      href: '/identityChecks/RPT19000083'
    },
    contact: {
      href: '/contacts/MKC13000122'
    },
    identityDocument1Type: {
      href: '/configuration/identityDocumentTypes/PP'
    },
    identityDocument2Type: {
      href: '/configuration/identityDocumentTypes/BB'
    }
  }
}
