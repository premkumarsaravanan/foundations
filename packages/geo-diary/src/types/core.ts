import Routes from '@/constants/routes'
import ActionTypes from '@/constants/action-types'
import { OnlineState } from '@/reducers/online'
import { ErrorState } from '@/reducers/error'
import { HomeState } from '@/reducers/home'
import { AppointmentsState } from '@/reducers/appointments'
import { AppointmentDetailState } from '@/reducers/appointment-detail'
import { AuthState } from '@/reducers/auth'
import { NextAppointmentState } from '@/reducers/next-appointment'
import { DirectionState } from '@/reducers/direction'
import {
  PagedResultAppointmentModel_,
  AppointmentModel,
  OfficeModel,
  PropertyModel,
  NegotiatorModel
} from '@reapit/foundations-ts-definitions'

export type ExtendedPagedResultAppointmentModel_ = Omit<PagedResultAppointmentModel_, '_embedded'> & {
  _embedded?: ExtendedAppointmentModel[]
}

export type ExtendedAppointmentModel = AppointmentModel & {
  offices?: OfficeModel[]
  property?: PropertyModel
  negotiators?: NegotiatorModel[]
}

export interface Action<T> {
  readonly type: ActionType
  readonly data: T
}

export interface ActionCreator<T> {
  readonly type: string
  (data: T): Action<T>
}

export interface StringMap {
  [key: string]: string
}

export type PartialRecord<K extends keyof any, T> = { [P in K]?: T }

export type RouteValue = keyof typeof Routes

export type ActionType = keyof typeof ActionTypes

export type FormState = 'PENDING' | 'DONE' | 'SUBMITTING' | 'ERROR' | 'SUCCESS'

export interface FetcherParams<T> {
  method: 'GET' | 'POST' | 'PUT' | 'DELETE'
  api: string
  url: string
  headers: StringMap
  isPrivate?: boolean
  body?: T
}

export interface ReduxState {
  online: OnlineState
  error: ErrorState
  home: HomeState
  direction: DirectionState
  appointments: AppointmentsState
  appointmentDetail: AppointmentDetailState
  auth: AuthState
  nextAppointment: NextAppointmentState
}
