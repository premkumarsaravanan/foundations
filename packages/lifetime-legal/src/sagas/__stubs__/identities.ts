export const identities = {
  _embedded: [
    {
      id: 'RPT19000083',
      contactId: 'MKC13000122',
      created: '0001-01-01T00:00:00',
      modified: '2020-01-02T09:29:36',
      checkDate: '0001-01-01T00:00:00',
      status: 'pass',
      negotiatorId: 'LJW',
      documents: [
        {
          typeId: 'PP',
          expiry: '2020-07-03T23:00:00',
          details: '123123',
          _links: {
            idDocumentType: {
              href: '/configuration/identityDocumentTypes/PP'
            }
          }
        },
        {
          typeId: 'BB',
          expiry: '2020-01-12T00:00:00',
          details: 'ID Reference',
          _links: {
            idDocumentType: {
              href: '/configuration/identityDocumentTypes/BB'
            }
          }
        }
      ],
      metadata: {
        primaryIdUrl: 'https://reapit-app-store-app-media.s3.eu-west-2.amazonaws.com/MKC13000122-123123(4).jpg',
        secondaryIdUrl: 'https://reapit-dev-app-store-media.s3.eu-west-2.amazonaws.com/MKC13000122-ID Reference.png',
        referralType: 'Estate Planning',
        dateCallClient: '2020-01-03T09:29:25',
        timeSelection: '01:00',
        clientType: 'Individual',
        isUKResident: 'No',
        placeMeet: 'Home Address'
      }
    },
    {
      id: 'RPT19000105',
      contactId: 'BED12001604',
      created: '0001-01-01T00:00:00',
      modified: '2019-11-01T09:22:37',
      checkDate: '0001-01-01T00:00:00',
      status: 'pass',
      negotiatorId: 'LJW',
      documents: [
        {
          typeId: 'CI',
          expiry: '2019-11-16T00:00:00',
          details: 'test',
          _links: {
            idDocumentType: {
              href: '/configuration/identityDocumentTypes/CI'
            }
          }
        }
      ],
      metadata: {
        primaryIdUrl: 'https://reapit-app-store-app-media.s3.eu-west-2.amazonaws.com/BED12001604-test.png',
        referralType: 'Estate Planning',
        dateCallClient: '2019-11-30T09:21:25',
        timeSelection: '14:22',
        clientType: 'Individual',
        isUKResident: 'Yes',
        placeMeet: 'Home Address'
      }
    }
  ],
  pageNumber: 1,
  pageSize: 50,
  pageCount: 2,
  totalCount: 2,
  _links: {
    self: {
      href:
        '/identityChecks/?PageNumber=1&PageSize=50&ContactId=MKC13000122&ContactId=BED12001604&ContactId=MKC12000004&ContactId=MKC07001329'
    },
    first: {
      href:
        '/identityChecks/?PageNumber=1&PageSize=50&ContactId=MKC13000122&ContactId=BED12001604&ContactId=MKC12000004&ContactId=MKC07001329'
    }
  }
}
