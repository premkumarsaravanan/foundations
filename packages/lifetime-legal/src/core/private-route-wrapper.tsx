import * as React from 'react'
import { Redirect, RouteComponentProps } from 'react-router-dom'
import { connect } from 'react-redux'
import { withRouter } from 'react-router'
import Routes from '../constants/routes'
import { ReduxState } from '@/types/core'
import { selectUserLoginStatus } from '@/selectors/auth'
import { Loader, AppNavContainer } from '@reapit/elements'
import { RefreshParams, getTokenFromQueryString } from '@reapit/cognito-auth'
import { Dispatch } from 'redux'
import { authSetRefreshSession } from '../actions/auth'
import Menu from '@/components/ui/menu'
import styles from '@/styles/index.scss?mod'

const { Suspense } = React

export interface PrivateRouteWrapperConnectActions {
  setRefreshSession: (refreshParams: RefreshParams) => void
}

export interface PrivateRouteWrapperConnectState {
  hasSession: boolean
  isDesktopMode: boolean
}

export type PrivateRouteWrapperProps = PrivateRouteWrapperConnectState &
  PrivateRouteWrapperConnectActions &
  RouteComponentProps & {
    path: string
  } & {
    children: React.ReactNode | React.ReactNodeArray
  }

export const PrivateRouteWrapper: React.FunctionComponent<PrivateRouteWrapperProps> = ({
  children,
  hasSession,
  isDesktopMode,
  setRefreshSession,
}: PrivateRouteWrapperProps) => {
  const desktopLogin = getTokenFromQueryString(location.search)

  if (desktopLogin && !isDesktopMode) {
    setRefreshSession(desktopLogin)
  }

  if (!hasSession) {
    return <Redirect to={Routes.LOGIN} />
  }

  return (
    <AppNavContainer>
      <div className={styles.navbar}>
        <Menu />
      </div>
      <Suspense fallback={<Loader body />}>{children}</Suspense>
    </AppNavContainer>
  )
}

const mapStateToProps = (state: ReduxState): PrivateRouteWrapperConnectState => ({
  hasSession: selectUserLoginStatus(state),
  isDesktopMode: state?.auth?.refreshSession?.mode === 'DESKTOP',
})

const mapDispatchToProps = (dispatch: Dispatch): PrivateRouteWrapperConnectActions => ({
  setRefreshSession: refreshParams => dispatch(authSetRefreshSession(refreshParams)),
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(PrivateRouteWrapper))
