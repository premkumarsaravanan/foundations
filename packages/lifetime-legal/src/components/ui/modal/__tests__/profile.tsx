import React from 'react'
import { shallow } from 'enzyme'
import { Profile, renderForm, filterCommunication, mapStateToProps, mapDispatchToProps } from '../profile'
import { contact } from '@/sagas/__stubs__/contact'
import { ReduxState } from '@/types/core'

describe('profile', () => {
  describe('renderForm', () => {
    it('should match snapshot', () => {
      const mockIsSubmitting = false
      const component = renderForm({ isSubmitting: mockIsSubmitting })
      const wrapper = shallow(<div>{component}</div>)
      expect(wrapper).toMatchSnapshot()
    })
  })
  describe('renderForm', () => {
    it('should match snapshot', () => {
      const mockIsSubmitting = false
      const component = renderForm({ isSubmitting: mockIsSubmitting })
      const wrapper = shallow(<div>{component}</div>)
      expect(wrapper).toMatchSnapshot()
    })
  })
  describe('filterCommunication', () => {
    it('should run correctly', () => {
      const expected = '01632 961556'
      const result = filterCommunication(contact.communications, 'Home')
      expect(result).toEqual(expected)
    })
    it('should return null', () => {
      const expected = null
      const result = filterCommunication(undefined, 'Home')
      expect(result).toEqual(expected)
    })
    it('should return null', () => {
      const expected = null
      const result = filterCommunication(undefined, 'test' as any)
      expect(result).toEqual(expected)
    })
  })
  describe('Profile', () => {
    const mockProps = {
      contact: contact,
      onSubmitHandler: jest.fn(),
      isSubmitting: false
    }
    const wrapper = shallow(<Profile {...mockProps} />)
    expect(wrapper).toMatchSnapshot()
  })

  describe('mapStateToProps', () => {
    it('should run correctly', () => {
      // @ts-ignore: only pick necessary props
      const mockState = {
        checklistDetail: {
          checklistDetailData: {
            contact
          }
        }
      } as ReduxState
      const result = mapStateToProps(mockState)
      expect(result).toEqual({
        contact,
        isSubmitting: false
      })
    })
    it('should run correctly', () => {
      const mockState = {} as ReduxState
      const result = mapStateToProps(mockState)
      expect(result).toEqual({
        contact: {},
        isSubmitting: false
      })
    })
  })
  describe('mapDispatchToProps', () => {
    it('should render correctly', () => {
      const mockDispatch = jest.fn()
      const { onSubmitHandler } = mapDispatchToProps(mockDispatch)
      onSubmitHandler && onSubmitHandler({})
      expect(mockDispatch).toBeCalled()
    })
    it('should render correctly', () => {
      const mockDispatch = jest.fn()
      const { onSubmitHandler } = mapDispatchToProps(mockDispatch)
      onSubmitHandler && onSubmitHandler({})
      expect(mockDispatch).toBeCalled()
    })
  })
})
