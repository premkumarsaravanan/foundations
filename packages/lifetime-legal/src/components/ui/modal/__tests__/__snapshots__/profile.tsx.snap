// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <Formik
    initialValues={
      Object {
        "dateOfBirth": null,
        "email": "schadwick512@rpsfiction.net",
        "forename": "Saoirse",
        "home": "01632 961556",
        "mobile": "07700 901556",
        "surname": "Chadwick",
        "title": "Ms",
        "work": "020 7946 1556",
      }
    }
    onSubmit={[MockFunction]}
  >
    <Component />
  </Formik>
</div>
`;

exports[`profile renderForm should match snapshot 1`] = `
<div>
  <Component />
</div>
`;

exports[`profile renderForm should match snapshot 2`] = `
<div>
  <Component />
</div>
`;
